### PR TITLE
Update to Kotlin 1.5.10

### PR DESCRIPTION
**Pull request by idanakav**
**Description**:
In order to update to Kotlin 1.5.10 we had to update other dependencies & tools as well:

1. Gradle -> [6.9](https://docs.gradle.org/6.9/release-notes.html)
2. Android tools & gradle plugin -> [4.2.0](https://developer.android.com/studio/releases/gradle-plugin#4-2-0)
3. Dokka to-> [1.4.32](https://kotlin.github.io/dokka/1.4.32/)
4. IntelliJ Gradle Plugin -> [1.0](https://lp.jetbrains.com/gradle-intellij-plugin/)
6. KotlinPoet -> [1.8.0](https://square.github.io/kotlinpoet/changelog/)
7. Proguard -> 6.2.2

A few other commits with code cleanup and deprecations are included, see any commit message for more details.

### CODE REVIEW

File: gradlew [hunk 1]
```diff
@@ -28,7 +44,7 @@ APP_NAME="Gradle"
 APP_BASE_NAME=`basename "$0"`
 
 # Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS=""
+DEFAULT_JVM_OPTS='"-Xmx64m" "-Xms64m"'
```
Comment (nikhilramakrishnan) [inline gradlew:47]: is this automatic?

File: intellij/ast/src/main/kotlin/motif/ast/intellij/IntelliJMethodParameter.kt [hunk 2]
```diff
@@ -31,7 +31,7 @@ class IntelliJMethodParameter(
         IntelliJType(project, substitutor.substitute(psiParameter.type))
     }
 
-    override val name: String by lazy { psiParameter.name!! }
+    override val name: String by lazy { psiParameter.name }
```
Comment (nikhilramakrishnan) [inline intellij/ast/src/main/kotlin/motif/ast/intellij/IntelliJMethodParameter.kt:34]: nullable?

**Review by nikhilramakrishnan — APPROVED** [review]

File: gradlew (hunk 1 above)
Comment (idanakav) [reply gradlew:47]: Yes, it's generated when the gradle wrapper is updated.

**Review by jbarr21 — APPROVED** [review]
+1 once CI is green

**Pull request merged**